Code/Editor/EditorFramework/CodeGen/CppProject.cpp

```diff
@@ -515,3 +901,241 @@ bool ezCppProject::IsBuildRequired()
+        if (RegGetValueW(hInstalledRoots, nullptr, L"KitsRoot10", RRF_RT_REG_SZ, nullptr, path.GetData(), &pathLengthInBytes) == ERROR_SUCCESS)
+        {
+          ezStringBuilder windowsSdkBinPath;
+          windowsSdkBinPath = ezStringWChar(path.GetData());
+          windowsSdkBinPath.MakeCleanPath();
+          windowsSdkBinPath.AppendPath("bin");
+
+          ezDynamicArray<ezFileStats> folders;
+          ezOSFile::GatherAllItemsInFolder(folders, windowsSdkBinPath, ezFileSystemIteratorFlags::ReportFolders);
+
+          folders.Sort([](const ezFileStats& a, const ezFileStats& b) { return a.m_sName > b.m_sName; });
+
+          for (const ezFileStats& folder : folders)
+          {
+            if (!folder.m_sName.StartsWith("10."))
+            {
+              continue;
+            }
+            rcPath = windowsSdkBinPath;
+            rcPath.AppendPath(folder.m_sName);
+            rcPath.AppendPath("x64/rc.exe");
+            if (ezOSFile::ExistsFile(rcPath))
+            {
+              break;
+            }
+            rcPath.Clear();
+          }
+        }
+      }
+    }
+    if (!rcPath.IsEmpty())
+    {
+      preferences->m_CompilerPreferences.m_RcCompiler = rcPath;
+    }
+  }
+
+  ezString clangVersion;
+
+  wchar_t* pProgramFiles = nullptr;
+  if (SUCCEEDED(SHGetKnownFolderPath(FOLDERID_ProgramFiles, KF_FLAG_DEFAULT, nullptr, &pProgramFiles)))
+  {
+    ezStringBuilder clangDefaultPath;
+    clangDefaultPath = ezStringWChar(pProgramFiles);
+    CoTaskMemFree(pProgramFiles);
+    pProgramFiles = nullptr;
+
+    clangDefaultPath.AppendPath("LLVM/bin/clang.exe");
+    clangDefaultPath.MakeCleanPath();
+    ezStringBuilder clangCppDefaultPath = clangDefaultPath;
+    clangCppDefaultPath.ReplaceLast(".exe", "++.exe");
+
+    ezStringView clangMajorSdkVersion = EZ_PP_STRINGIFY(__clang_major__) ".";
+    if (TestCompilerExecutable(clangDefaultPath, &clangVersion).Succeeded() && TestCompilerExecutable(clangCppDefaultPath).Succeeded() && clangVersion.StartsWith(clangMajorSdkVersion))
+    {
+      ezStringBuilder clangNiceName;
+      clangNiceName.Format("Clang (system default = {})", clangVersion);
+      s_MachineSpecificCompilers.PushBack({clangNiceName, ezCompiler::Clang, clangDefaultPath, clangCppDefaultPath, false});
+    }
+  }
+#  endif
+#endif
+
+
+#if EZ_ENABLED(EZ_PLATFORM_LINUX)
+  AddCompilerVersions(s_MachineSpecificCompilers, ezCppProject::GetSdkCompiler(), ezCppProject::GetSdkCompilerMajorVersion());
+#endif
+
+#if EZ_ENABLED(EZ_COMPILER_CLANG)
+  s_MachineSpecificCompilers.PushBack({"Clang (Custom)", ezCompiler::Clang, "", "", true});
+#endif
+
+#if EZ_ENABLED(EZ_PLATFORM_LINUX) && EZ_ENABLED(EZ_COMPILER_GCC)
+  s_MachineSpecificCompilers.PushBack({"Gcc (Custom)", ezCompiler::Gcc, "", "", true});
+#endif
+
+  if (preferences->m_CompilerPreferences.m_eCompiler != sdkCompiler)
+  {
+    s_MachineSpecificCompilers.PushBack(
+      {"SDK Incompatible previously configured compiler",
```

Comment:
Is this missing a "with" in the text ?

Reply:
No. This piece of code auto detects available compilers on the system. If the currently serialized compiler is different from the compiler the SDK was built with we add a dummy compiler which exactly represents the serialized state. This is done so the user doesn't loose the configuration upon exiting the editor no matter if that compiler still exists or not. So what this means is "A compiler that was previously configured but is now incompatible with the current SDK build" -> "SDK incompatible previously configured compiler"

Reply:
This text still doesn't make sense to me. Wouldn't it then be "SDK COMPATIBLE previously configured compiler" ? As in the previously configured compiler WAS compatible (not INcompatible).
Maybe it would be better to rephrase this entirely: "Missing: Compiler the SDK was built with"
Or something similar.

Reply:
Imagine the following scenario:

1) User builds Editor with Vs2022. Goes into preferences and selects VS2022 as compiler. User closes Editor.
2) User builds Editor with Clang. Opens Editor goes into preferences. The previously selected VS2022 compiler is now no longer compatible with the Editor. This is why there will now be a entry in the dropdown saying "SDK incompatible previously configured compiler". The user can either choose to just keep that value. For example if the user only wanted to quickly test something with the clang Editor without needing C++ Plugins. This way the compiler preferences don't change. Or the user can switch from the incompatible compiler to a Clang version that was auto detected.
3) If the user didn't change the compiler in preferences and opens the Editor build with VS2022 again, the dropdown will again read VS2022, because now the compiler is compatible with the Editor again.

The same can also happen on linux with clang / gcc. Although most likely most users will never see this option because they will always build with the same compiler.